### PR TITLE
Move to UI directory

### PR DESCRIPTION
**Pull request by ylavoie**
Move all UI related files from the root to the UI directory to allow proper separation between the server amd UI applications.

### CODE REVIEW

File: UI/webpack.config.js [hunk 1]
```diff
@@ -406,15 +409,15 @@ if (TARGET !== "readme") {
             topLevelAwait: true
         },
 
-        context: path.join(__dirname, "UI"),
+        context: path.join(__dirname, ""),
```
Comment (ehuelsmann) [inline UI/webpack.config.js:412]: now, it's no longer needed to join?

File: UI/webpack.config.js [hunk 2]
```diff
@@ -494,18 +497,18 @@ if (TARGET !== "readme") {
             },
             static: [
                 {
-                    directory: path.join(__dirname, "/UI"),
+                    directory: path.join(__dirname, ""),
```
Comment (ehuelsmann) [inline UI/webpack.config.js:500]: No need to join?

File: UI/webpack.config.js [hunk 3]
```diff
@@ -494,18 +497,18 @@ if (TARGET !== "readme") {
...
                     publicPath: '/'
                 },
                 {
-                    directory: path.join(__dirname, "/UI"),
+                    directory: path.join(__dirname, ""),
```
Comment (ehuelsmann) [inline UI/webpack.config.js:504]: same: no need to join?

File: UI/webpack.config.js [hunk 4]
```diff
@@ -172,21 +172,21 @@ if (TARGET !== "readme") {
...
         }
     };
 
     const DojoWebpackPluginOptions = {
-        loaderConfig: require("./UI/js-src/lsmb/webpack.loaderConfig.js"),
-        environment: { dojoRoot: "UI/js" }, // used at run time for non-packed resources (e.g. blank.gif)
+        loaderConfig: require("./js-src/lsmb/webpack.loaderConfig.js"),
+        environment: { dojoRoot: "js" }, // used at run time for non-packed resources (e.g. blank.gif)
         buildEnvironment: { dojoRoot: "node_modules" }, // used at build time
-        locales: getPOFilenames("locale/po", ".po"),
```
Comment (ehuelsmann) [inline UI/webpack.config.js:189]: why does moving everything to UI/ cause the extension to change?

File: Makefile [hunk 5]
```diff
@@ -83,29 +83,29 @@ dbdocs:
...
-	$(DOCKER_CMD) npm run lint:js:fix
+	$(DOCKER_CMD) cd UI && npm run lint:js:fix
 else
-	$(DOCKER_CMD) npm run lint:js
+	$(DOCKER_CMD) cd UI && npm run lint:js
 endif
 
 readme: npm_install
-	$(DOCKER_CMD) npm run readme
+	$(DOCKER_CMD) cd UI && npm run readme
```
Comment (ehuelsmann) [inline Makefile:108]: Doesn't this cause the "npm" command to be run outside of the docker environment?

File: Makefile [hunk 6]
```diff
@@ -83,29 +83,29 @@ dbdocs:
...
-	$(DOCKER_CMD) npm run build:dev
+	$(DOCKER_CMD) cd UI && npm run build:dev
 
 jslint:
 ifneq ($(origin FIX),undefined)
-	$(DOCKER_CMD) npm run lint:js:fix
+	$(DOCKER_CMD) cd UI && npm run lint:js:fix
 else
-	$(DOCKER_CMD) npm run lint:js
+	$(DOCKER_CMD) cd UI && npm run lint:js
```
Comment (ehuelsmann) [inline Makefile:104]: Doesn't this cause the "npm" command to be run outside of the docker environment?

File: Makefile [hunk 7]
```diff
@@ -83,29 +83,29 @@ dbdocs:
...
+	$(DOCKER_CMD) cd UI && npm run build
 
 jsdev: npm_install
-	$(DOCKER_CMD) npm run build:dev
+	$(DOCKER_CMD) cd UI && npm run build:dev
 
 jslint:
 ifneq ($(origin FIX),undefined)
-	$(DOCKER_CMD) npm run lint:js:fix
+	$(DOCKER_CMD) cd UI && npm run lint:js:fix
```
Comment (ehuelsmann) [inline Makefile:102]: Doesn't this cause the "npm" command to be run outside of the docker environment?

File: Makefile [hunk 8]
```diff
@@ -83,29 +83,29 @@ dbdocs:
...
-	$(DOCKER_CMD) npm run build:dev
+	$(DOCKER_CMD) cd UI && npm run build:dev
 
 js: npm_install
-	$(DOCKER_CMD) npm run build
+	$(DOCKER_CMD) cd UI && npm run build
 
 jsdev: npm_install
-	$(DOCKER_CMD) npm run build:dev
+	$(DOCKER_CMD) cd UI && npm run build:dev
```
Comment (ehuelsmann) [inline Makefile:98]: Doesn't this cause the "npm" command to be run outside of the docker environment?

File: UI/webpack.config.js (hunk 4 above)
Comment (ylavoie) [reply UI/webpack.config.js:189]: We can grab the locales through `locale/*.PO` or `UI/src/locales/*.json`. In order to become less tied to the server, I chose to enumerate JSON instead.

File: Makefile (hunk 5 above)
Comment (ylavoie) [reply Makefile:108]: No

File: Makefile (hunk 7 above)
Comment (ylavoie) [reply Makefile:102]: No

File: Makefile (hunk 6 above)
Comment (ylavoie) [reply Makefile:104]: No

File: UI/webpack.config.js (hunk 2 above)
Comment (ylavoie) [reply UI/webpack.config.js]: Fixed

File: UI/webpack.config.js (hunk 1 above)
Comment (ylavoie) [reply UI/webpack.config.js]: Fixed

File: Makefile (hunk 8 above)
Comment (ehuelsmann) [reply Makefile:98]: I don't get that. How can you be so sure? Do you have npm installed inside and outside the container? Or just inside?

Comment (ylavoie) [reply Makefile]: Fixed.

**Review by ehuelsmann — APPROVED** [review]

**Pull request merged**